extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -634,21 +628,24 @@ export class Bind {
   }
 
   /**
-   * Applies expression results to DOM.
+   * Applies expression results to the DOM.
    * @param {Object<string, ./bind-expression.BindExpressionResultDef>} results
+   * @param {boolean=} opt_fromAmpState
    * @private
    */
-  apply_(results) {
+  apply_(results, opt_fromAmpState) {
     const applyPromises = [];
     this.boundElements_.forEach(boundElement => {
-
       const {element, boundProperties} = boundElement;
+      // If this "apply" round is triggered by an <amp-state> mutation,
+      // ignore updates to <amp-state> element to prevent update cycles.
+      if (opt_fromAmpState && element.tagName === 'AMP-STATE') {
```

Comment:
I know we've got similar checks elsewhere in the file, but is the casing of `tagName` guaranteed to be uppercase?